contracts/VotingMachines/GenesisProtocol.sol

```diff
@@ -100,6 +100,8 @@ contract GenesisProtocol is IntVoteInterface,UniversalScheme {
     uint public proposalsCnt; // Total number of proposals
     mapping(address=>uint) public orgBoostedProposalsCnt;
     StandardToken public stakingToken;
+    mapping(address=>uint[]) public proposalsExpiredTimes; //proposals expired times
```

Comment:
Should also work when the DAO changes params (unregisters and registers). Maybe use a single [Heap](https://en.wikipedia.org/wiki/Heap_(data_structure)) per DAO.